raiden/blockchain/filters.py

```diff
@@ -127,13 +127,16 @@ def _do_get_new_entries(
         self._last_block = block_specification_to_number(block=to_block, web3=self.web3)
         return result
 
+    def from_block_number(self) -> BlockNumber:
+        filter_from_number = block_specification_to_number(
+            block=self.filter_params.get("fromBlock", GENESIS_BLOCK_NUMBER), web3=self.web3
+        )
+        return BlockNumber(max(filter_from_number, self._last_block + 1))
+
     def get_new_entries(self, target_block_number: BlockNumber) -> List[BlockchainEvent]:
         with self._lock:
             result: List[BlockchainEvent] = []
-            filter_from_number = block_specification_to_number(
-                block=self.filter_params.get("fromBlock", GENESIS_BLOCK_NUMBER), web3=self.web3
-            )
-            from_block_number = max(filter_from_number, self._last_block + 1)
+            from_block_number = int(self.from_block_number())
```

Comment:
Why do we need to cast to `int`, here?

Reply:
Mypy was nagging about the `+=` assignment below. I put a `# type: ignore` in that line instead, so it's clearer